docs/rfcs/template.md

```diff
@@ -0,0 +1,19 @@
+# IMEX Design Document (RFC)
+
+## Summary
+Short description of proposed solution.
+
+## Motivation
+Explain motivation for the proposed idea.
+If you are proposing an optimization pass, please explain the applicable scenario and expected performance gain. It needs to address below criteria:
+- The new dialect needs to introduce unique features which we don’t think upstream dialect has interests to cover.
+- The new dialect should facilitate certain compile-time optimizations which otherwise cannot or very hard to achieve. In other words, there should be compile-time optimizations which requires the computation graph to be represented using the new dialects. If there is no compile-time optimization required, the functions could be represented  as a standardized MLIR module to facilitate code reuse.
```

Comment:
I am confused: This defines criteria for _optimizations passes_ but the criteria talk about characteristics of the _dialects_. I am not sure if the distinction between passes and dialects is intended, but if it is then this is unclear.

@Jianhui-Li @drprajap What is a "standard MLIR module"?

Reply:
Above criteria should be applicable to both pass and dialect. there is no distinction intended here. Standard MLIR module could be referred to as high level function that utilizes existing/upstream dialects/passes but provides certain flow of operations enclosed within a function. One example that I can think of is something similar as described [here](https://mlir.llvm.org/docs/LangRef/#high-level-structure)